### PR TITLE
Add documentation about AutoFeat

### PR DESCRIPTION
**Pull request by Hiba-Alili**

### CODE REVIEW

**Review by andrewssobral — APPROVED** [review]
Perfect, thank you @Hiba-Alili

**Comment by mboussaa** [comment]
@andrewssobral I was reviewing this documentation when you merged it! Could you please wait a bit for a second review next time before merging it? Thx
@Hiba-Alili

File: src/docs/PML/PMLUserGuide.adoc
Comment (mboussaa) [inline src/docs/PML/PMLUserGuide.adoc:1569]: - machine learning => Machine Learning? (I am not sure)
- not only depends => depends not only

File: src/docs/PML/PMLUserGuide.adoc
Comment (mboussaa) [inline src/docs/PML/PMLUserGuide.adoc:1571]: there are various tasks we require to perform as a part of the data preparation
=> 
we require to perform various tasks for data preparation or as part of data preparation

Comment (mboussaa) [reply src/docs/PML/PMLUserGuide.adoc:1571]: Encoding categorical data is one of such tasks which is considered crucial
=>
One of the most crucial tasks is 'categorical data encoding' (this is just a suggestion)

Comment (mboussaa) [reply src/docs/PML/PMLUserGuide.adoc:1571]: As we know, most of the data in real life come with categorical string values and most of the machine learning models basically perform mathematical operations. But the harsh truth is that mathematics is totally dependent on numbers. So in short we can say that most of the machine learning models only accept numerical variables, not strings and these numbers can be float or integer. Thereafter, preprocessing and encoding the categorical variables becomes a necessary step, such we need to convert these categorical variables to numbers which can help in predicting the outcomes in a machine learning task.

=>

In real life, data commonly come with categorical string values and most of the time machine learning models perform mathematical operations. However, the harsh truth is that mathematics is totally dependent on numbers. As a matter of fact, we can say that most of the machine learning models only accept numerical variables (generally floats or integers) and not strings. Hence, preprocessing and encoding the categorical variables become a crucial step to convert these variables to numbers that can help in predicting the results in a machine learning task.

File: src/docs/PML/PMLUserGuide.adoc
Comment (mboussaa) [inline src/docs/PML/PMLUserGuide.adoc:1575]: for the proper working of the model
to refactor as I didn't understant

Comment (mboussaa) [reply src/docs/PML/PMLUserGuide.adoc:1575]: can vary => can impact or influence u mean ?

File: src/docs/PML/PMLUserGuide.adoc
Comment (mboussaa) [inline src/docs/PML/PMLUserGuide.adoc:1588]: As we said earlier, => As we already mentioned, 
should we use ML everywhere? or machine learning or Machine Learning

Comment (mboussaa) [reply src/docs/PML/PMLUserGuide.adoc:1588]: The produced results by the model vary depending on the used encoding technique.

File: src/docs/PML/PMLUserGuide.adoc
Comment (mboussaa) [inline src/docs/PML/PMLUserGuide.adoc:1577]: do you have a reference ? for these techniques ?

File: src/docs/PML/PMLUserGuide.adoc
Comment (mboussaa) [reply src/docs/PML/PMLUserGuide.adoc:1588]: Thus, the hardest part of categorical encoding can sometimes be finding the right categorical encoding method. 
to refactor, hard to read

File: src/docs/PML/PMLUserGuide.adoc
Comment (mboussaa) [inline src/docs/PML/PMLUserGuide.adoc:1590]: dedicated to the analysis of the performance of categorical encoding approaches applied to different datasets.

Comment (mboussaa) [reply src/docs/PML/PMLUserGuide.adoc:1590]: the best suited method

Comment (mboussaa) [reply src/docs/PML/PMLUserGuide.adoc:1590]: for your data => scientists data isn't better? I dunnow

Comment (mboussaa) [reply src/docs/PML/PMLUserGuide.adoc:1590]: This can help the data scientist to start encoding smarter. 
I think to remove or refactor. 
It looks like they are stupid now but will be smarter haha :D

File: src/docs/PML/PMLUserGuide.adoc
Comment (mboussaa) [inline src/docs/PML/PMLUserGuide.adoc:1594]: I think this will be trydev2 when it is on trydev2 right? not try

File: src/docs/PML/PMLUserGuide.adoc
Comment (mboussaa) [inline src/docs/PML/PMLUserGuide.adoc:1598]: isn't better to show how autofit works from WE only? Is there any need to do that from Studio?

File: src/docs/PML/PMLUserGuide.adoc
Comment (mboussaa) [inline src/docs/PML/PMLUserGuide.adoc:1606]: workflow's variables => workflow variables

File: src/docs/PML/PMLUserGuide.adoc
Comment (mboussaa) [inline src/docs/PML/PMLUserGuide.adoc:1631]: (optional) the endpoint will have a name, u can imagine a name for now and integrate it to the doc

File: src/docs/PML/PMLUserGuide.adoc
Comment (mboussaa) [inline src/docs/PML/PMLUserGuide.adoc:1629]: Open the https://try.activeeon.com/automation-dashboard/#/portal/workflow-execution[Workflow Execution Portal].

File: src/docs/PML/PMLUserGuide.adoc
Comment (mboussaa) [inline src/docs/PML/PMLUserGuide.adoc:1633]: contains three tabs that we describe in the following sections.

File: src/docs/PML/PMLUserGuide.adoc
Comment (mboussaa) [inline src/docs/PML/PMLUserGuide.adoc:1637]: Only => only

Comment (mboussaa) [reply src/docs/PML/PMLUserGuide.adoc:1637]: very large isn't better?

File: src/docs/PML/PMLUserGuide.adoc
Comment (mboussaa) [inline src/docs/PML/PMLUserGuide.adoc:1647]: values to at the end if I well understand

File: src/docs/PML/PMLUserGuide.adoc
Comment (mboussaa) [inline src/docs/PML/PMLUserGuide.adoc:1662]: .There => . There

File: src/docs/PML/PMLUserGuide.adoc
Comment (mboussaa) [inline src/docs/PML/PMLUserGuide.adoc:1666]: Label column is the feature about which we want to gain a deeper understanding. 
to refacor

File: src/docs/PML/PMLUserGuide.adoc
Comment (mboussaa) [inline src/docs/PML/PMLUserGuide.adoc:1668]: numeric => numerical?

Comment (mboussaa) [reply src/docs/PML/PMLUserGuide.adoc:1668]: the method best suitedv same as my previous remark

Comment (mboussaa) [reply src/docs/PML/PMLUserGuide.adoc:1668]: power => ability

Comment (mboussaa) [reply src/docs/PML/PMLUserGuide.adoc:1668]: depending on the selected method?

Comment (mboussaa) [inline src/docs/PML/PMLUserGuide.adoc:1668]: can he know which best method the algo has selected for him? when auto is enabled ?

File: src/docs/PML/PMLUserGuide.adoc
Comment (mboussaa) [inline src/docs/PML/PMLUserGuide.adoc:1680]: what are the metrics used to compare the results of the different methods?

**Comment by andrewssobral** [comment]
@mboussaa ok sorry

File: src/docs/PML/PMLUserGuide.adoc
Comment (Hiba-Alili) [reply src/docs/PML/PMLUserGuide.adoc:1577]: Yes, i added the reference for the python library (category encoders) that proposed an implementation of those techniques. Do you suggest to add also the reference for each of those techniques?

File: src/docs/PML/PMLUserGuide.adoc
Comment (Hiba-Alili) [reply src/docs/PML/PMLUserGuide.adoc:1647]: I'm not sure I understand this comment correctly. Is that abt adding values to the end of line 1647?

File: src/docs/PML/PMLUserGuide.adoc
Comment (Hiba-Alili) [reply src/docs/PML/PMLUserGuide.adoc:1668]: Not at this stage, we're planning to add it as a futur enhancement.

File: src/docs/PML/PMLUserGuide.adoc
Comment (Hiba-Alili) [reply src/docs/PML/PMLUserGuide.adoc:1680]: For now, the user can check himself the results. We may add some comparison metrics in a futur work.

File: src/docs/PML/PMLUserGuide.adoc
Comment (Hiba-Alili) [reply src/docs/PML/PMLUserGuide.adoc:1598]: A good catch, i'm gonna check and update it accordingly after adding the endpoint on WE. Thank you mohamed :)

**Pull request merged**
